src/components/modals/EntityCreatorEditor/EntityCreatorContainer.tsx

```diff
@@ -337,9 +363,53 @@ class Container extends React.Component<Props, ComponentState> {
             entityNameVal: isPrebuilt ? getPrebuiltEntityName(obj.text) : prevState.isPrebuilt ? "" : prevState.entityNameVal,
         }))
     }
+
+    onDeleteEnum = (enumValue: CLM.EnumValue) => {
+        if (this.isEnumRequiredForActions(enumValue)) {
+            this.setState({
+                deleteEnumCheck: enumValue
+            })
+            return
+        }
+        else {
+            this.deleteEnum(enumValue)
+        }
+    }
+
+    deleteEnum(enumValue: CLM.EnumValue): void {
+        let index = this.state.enumValues.indexOf(enumValue)
+        if (index >= 0) {
+            let enumValues = [...this.state.enumValues]
+            enumValues[index] = null
+            this.setState({enumValues})
+        }
+    }
+
     onChangedEnum = (index: number, value: string) => {
         let enumValues = [...this.state.enumValues]
-        enumValues[index] = value.toUpperCase()
+        const newValue = value.toUpperCase().trim()
+        let enumValue = enumValues[index]
+
+        if (newValue.length > 0) {    
+            // Create new EnumValue if needed 
+            if (!enumValue) {
+                enumValues[index] = { enumValue: newValue }
+            }
+            // Otherwise set
+            else {
+                enumValue.enumValue = newValue
```

Comment:
Hmm this again.  `enumValue.enumValue`  

I think I understand that the first one is referring to the object and the second is the string? 
If so we should try to call them that, especially when they're mixed so people don't get confused. Like `enumValueObject.value`?

Reply:
good suggestion